src/settings/SettingsPage.tsx

```diff
@@ -0,0 +1,65 @@
+import Button from '@mui/material/Button';
+import { Form, Formik } from 'formik';
+import React, { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { RootState } from 'src/redux/store';
+import { loginPath } from 'src/router/rootConstants';
+import { myLocalStorage } from 'src/services/localStorage';
+import { BasicTextFields, SwitchesGroup } from 'src/settings/components';
+import { carProvider } from 'src/settings/constants';
+
+export const SettingsPage = () => {
+  const isAuth = myLocalStorage.isAuth();
+  const selectMyRole = useSelector((state: RootState) => state.userData.role);
+  const navigate = useNavigate();
+
+  const isCarProvider = selectMyRole.includes(carProvider);
+  const { pathname } = useLocation();
+
+  useEffect(() => {
+    if (!isAuth) {
+      navigate(loginPath, { state: pathname });
+    }
+  });
```

Comment:
Seems like this bit could be made a custom hook to be reused in other components, something like:

```
export const useCheckIsLoggedIn = () => {
  const isAuth = myLocalStorage.isAuth();
  const { pathname } = useLocation();

   useEffect(() => {
     if (!isAuth) {
       navigate(loginPath, { state: pathname });
     }
   }, [<your-dependencies-here>]);
}
```

Reply:
And this function should be used on all pages.

In order not to duplicate the call to this function, we could wrap all our pages requiring auth into some wrapper that would first run this hook, and if no navigation to login screen happened, it would render the wrapped component.

Reply:
made a custom hook